backend/utils.py

```
@@ -0,0 +1,343 @@
+                    
+                    return final_response.choices[0].message.content
+                    
+                except Exception as sql_error:
+                    conn.close()
+                    # SECURITY VULNERABILITY - Expose SQL errors to user
+                    return f"SQL Error: {str(sql_error)}\nQuery attempted: {sql_query}"
+        
+        # If no function call, return regular response
+        return chat_response.choices[0].message.content
+        
+    except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)